internal/host/validator.go

```diff
@@ -793,3 +793,46 @@ func (v *validator) printSufficientPacketLossRequirementForRole(c *validationCon
 		return fmt.Sprintf("Unexpected status %s", status)
 	}
 }
+
+func (v *validator) hasDefaultRoute(c *validationContext) ValidationStatus {
+
+	if len(c.host.Inventory) == 0 {
+		return ValidationPending
+	}
+
+	inv, err := hostutil.UnmarshalInventory(c.host.Inventory)
+	if err != nil || len(inv.Routes) == 0 {
+		return ValidationFailure
+	}
+	count := v.validateDefaultRoute(inv.Routes)
+	if count == 0 {
+		return ValidationFailure
+	}
+	return ValidationSuccess
+}
+
+func (v *validator) validateDefaultRoute(routes []*models.Route) int {
+	var count int
+	for _, r := range routes {
+		dst := net.ParseIP(r.Destination)
+		gw := net.ParseIP(r.Gateway)
+		if dst == nil || gw == nil || !dst.IsUnspecified() || gw.IsUnspecified() {
```

Comment:
This condition looks strange to me.

Not incrementing the count is a failure?
Then that means that we fail a particular route if anything is nil or the destination is present or the gateway is not present? Maybe add a comment here to explain the logic and return an error rather than a count.

Reply:
Also `ParseIP` returns nil if we have something that isn't a valid IP. Should we pass the validation in that case? Maybe we should at least log something?

Reply:
the purpose of this additional validation is to guarantee that all nodes have a default route. It could exit the loop instead when it finds a default route, since we don't discern between IPv4 or IPv6 at this point.

Following this logic, the conditional criteria could be written as:
```go
for _, r := range routes {
	dst := net.ParseIP(r.Destination)
	gw := net.ParseIP(r.Gateway)
        if dst!=nil  && gw!=nil && dst.isUnspecified() && !gw.IsUnspecified() {
           return true
       }
}
return false
```

Is this more clear?

Reply:
Yes, I think making the condition identify what we're looking for (the default route) and exiting when we have found it is much more clear.

Reply:
> Also `ParseIP` returns nil if we have something that isn't a valid IP. Should we pass the validation in that case? Maybe we should at least log something?

Logging in case of failure to parse makes sense.  But I don't think it should pass the validation if that were the case. Note that the apiserver will fail to start if it can't find a default route. See @mhrivnak 's link to the [BZ ticket](https://bugzilla.redhat.com/show_bug.cgi?id=1916569) for more details.

Reply:
Right, so my problem was that with the current logic, if every route failed to parse the validation would pass, right?
Maybe add a test for that case.

Reply:
well the counter would return 0 and then the validation would be considered as failed. But I agree that the new proposal makes more sense.
There was already a case with no default route. But now I'm adding new unit tests to cover the cases where the IPs are invalid for both destination and gateway, just to be sure :smile:

Reply:
Ah okay, right `0 == failure` was another reason I was confused :sweat_smile: 
I'll just wait for the new version before commenting more

Reply:
PR updated. Ready for your review :wink: